### ISSUE TITLE
Add wrapped stacktraces to RFC6749Error

### ISSUE THREAD

**Issue by beornf** [ory/fosite#577]
**Describe the bug**

Errors no longer include stacktraces due to regression in https://github.com/ory/fosite/commit/620d4c148307f7be7b2674fe420141b33aef6075.

**To Reproduce**

Compare attached stacktraces of `errors.WithStack(fosite.ErrUnknownRequest)` against `errorsx.WithStack(fosite.ErrUnknownRequest)`. The error variable `fosite.ErrUnkownRequest` does not have a stacktrace however it is not wrapped due to having a `StackTrace()` method which returns an empty slice.

**Expected behavior**

The error value `errorsx.WithStack(fosite.ErrUnknownRequest)` would return a wrapped error with attached stacktrace.

**Comment by aeneasr** [comment]
Ah nice catch, we could probably check if the stack trace is empty and then wrap it?

**Comment by beornf** [comment]
Yeah in that case the fix might need to be done in the `ory/x` package.

**Comment by aeneasr** [comment]
SGTM - would you be up for a PR? :)

**Comment by beornf** [comment]
Yeah I'll write one soon and link it to this issue.
